cmd/wfnconvert/options.go

```diff
@@ -0,0 +1,199 @@
+			return fmt.Errorf("failed to parse fields from %q: %v", s, err)
+		}
+		if i64 <= 0 {
+			return fmt.Errorf("illegal CSV index %d: %q", i64, s)
+		}
+		*f = append(*f, int(i64)-1)
+	}
+	return nil
+}
+
+// String is part of flag.Value interface
+func (f *intFields) String() string {
+	var sb strings.Builder
+	for i, n := range *f {
+		if i != 0 {
+			sb.WriteByte(',')
+		}
+		fmt.Fprintf(&sb, "%d", n+1)
+	}
+	return sb.String()
+}
+
+type strFields map[string]bool
+
+// Set is part of flag.Value interface
+func (f *strFields) Set(s string) error {
+	if *f == nil {
+		*f = make(strFields)
+	}
+	for _, s := range strings.Split(s, ",") {
+		(*f)[s] = true
+	}
+	return nil
+}
+
+// String is part of flag.Value interface
+func (f *strFields) String() string {
+	var sb strings.Builder
+	i := 0
+	for k := range *f {
+		if i != 0 {
+			sb.WriteByte(',')
+		}
+		sb.WriteString(k)
+		i++
+	}
+	return sb.String()
+}
+
+type options struct {
+	outBinding       string
+	attributes       strFields
+	invertAttributes bool
+	any2na, na2any   bool
+	csvFields        intFields
+	csvComma         string
+}
+
+func (o *options) addFlags() {
+	flag.StringVar(&o.outBinding, "b", "fstr", "ouput bindings, one of\n"+
+		"'uri'\te.g. cpe:/a:foo:bar:1.1)\n"+
+		"'fstr'\te.g. cpe:2.3:a:foo:bar:1.1:*:*:*:*:*:*:*)\n"+
+		"'str'\te.g. "+`wfn:[part="a",vendor="foo",product="bar", version="1\.1"])`+"\n")
+	flag.Var(&o.attributes, "attr", "comma-separated list of WFN attributes to manipulate; CPE 2.3 defines following attributes:\n"+
+		"part, vendor, product, version, update, edition, language, sw_edition, target_sw, target_hw, other\n"+
+		"additionally a special value 'all' is accepted")
+	flag.BoolVar(&o.invertAttributes, "v", false, "invert attributes: process attributes not matched by -v flag")
+	flag.BoolVar(&o.any2na, "any2na", false, "assign logical value N/A to all attributes with logical value ANY")
+	flag.BoolVar(&o.na2any, "na2any", false, "assign logical value ANY to all attributes with logical value N/A")
+	flag.Var(&o.csvFields, "csv", "comma-separated list of fields to read CPEs from, starting at 1;\n"+
+		"assumes input as CSV, fields not specified in the list are passed unchanged")
+	flag.StringVar(&o.csvComma, "d", ",", "if csv flag is set, sets the CSV delimiter, otherwise this flag is ignored")
+}
+
+func (o *options) validate() error {
+	if err := validateAttrNames(o.attributes); err != nil {
+		return err
+	}
+
+	if o.any2na && o.na2any {
+		return fmt.Errorf("-any2na and -na2any flags are mutually exclusive")
+	}
+
+	switch o.outBinding {
+	case "uri", "fstr", "str":
+		break
+	default:
+		return fmt.Errorf("-b: invalid value %q", o.outBinding)
+	}
+
+	return nil
+}
+
+func (o *options) shouldProcess(name string) bool {
+	do := o.attributes["all"] || o.attributes[name]
+	if o.invertAttributes {
+		return !do
+	}
+	return do
+}
+
+func (o *options) processAttributes(attr *wfn.Attributes, f func(*string) error) error {
+	if o.shouldProcess("part") {
```

Comment:
can you make this something like:
```
for name, addr := range map[string]*string{
  "part": &attr.Part,
  ...
} {
  if o.shouldProcess(name) {
    if err := f(addr); err != nil {
      return err
    }
  }
}
```

Reply:
Now I see you did that already in validate names

p.s. I was actually thinking of adding something like this to attributes themselves
like:
```
func (a *Attributes) Traverse(f func(*string) error) error
```

Reply:
yes, no?

Reply:
no.